test/functional/tests.py

```diff
@@ -648,5 +739,86 @@ def test_bitswap_unwant(self):
         result = self.api.bitswap_unwant(key='QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V')
         self.assertTrue(result is not None)
 
+# Run test for `.shutdown()` only as the last test in CI environments – it would be to annoying
+# during normal testing
+@skipIfOffline()
+@skipUnlessCI()
+@pytest.mark.last
+class IpfsApiShutdownTest(unittest.TestCase):
+    def setUp(self):
+        self.api = ipfsapi.Client()
+        self.pid = int(os.environ["PY_IPFSAPI_TEST_DAEMON_PID"])
+    
+    @staticmethod
+    def _pid_exists(pid):
+        """
+        Check whether pid exists in the current process table
+
+        Source: https://stackoverflow.com/a/23409343/277882
```

Comment:
I shouldn't be surprised by how gross this solution is, but somehow I still am. Thanks for finding a way!